xled/control.py

```diff
@@ -410,6 +416,36 @@ def set_timer(self, time_on, time_off, time_now=None):
         required_keys = [u"code"]
         assert all(key in app_response.keys() for key in required_keys)
 
+    def send_realtime_frame(self, leds_number, bytes_per_led, data):
+        """
+        Sends a realtime frame. The mode must have been set to 'rt'.
+
+        :param int leds_number: the number of leds in a frame
+        :param int bytes_per_led: the number of bytes per led (3 or 4)
+        :param bytearray data: byte array containing the raw frame data
+        :rtype: None
+        """
+        data_size = leds_number*bytes_per_led
+        assert len(data) == data_size
+        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as sock:
+            if data_size < 900 and leds_number < 256:
+                # Send single frame
+                packet = bytearray(b'\x01')
+                packet.extend(base64.b64decode(self.session.access_token))
+                packet.extend(bytes([leds_number]))
+                packet.extend(data)
+                sock.sendto(packet, (self.host, REALTIME_UDP_PORT_NUMBER))
+            else:
+                # Send multi frame
+                packet_size = 900//bytes_per_led
+                for i in range(0, math.ceil(data_size/packet_size)):
+                    packet_data = data[:(900//bytes_per_led)]
```

Comment:
Oops!  The packet data it sends is always the same initial segment.  This expression needs to depend on `i`.

I can't test this case unfortunately because I have 250 x RGB lights.

Also, there's another issue with this where it does all these bytearray concatenations on every frame.  Most of the packet is the same every time it's sent, so an optimization would be to have a fixed sized packet and simply copy in the data.

(Sorry for the single comments one at a time, I'm writing code as I go.  :-)